string finger
int64 velocity

